signed/verifiers.go

```diff
@@ -157,6 +163,9 @@ type RSAPyCryptoVerifier struct{}
 // with PyCrypto.
 func (v RSAPyCryptoVerifier) Verify(key data.PublicKey, sig []byte, msg []byte) error {
 	digest := sha256.Sum256(msg)
+	if key.Algorithm() != data.RSAKey {
+		return ErrInvalidKeyType{}
+	}
 
 	k, _ := pem.Decode([]byte(key.Public()))
```

Comment:
"legacy" just means existing code. If some other system used shorter keys (as terrible as that would be), it doesn't make sense for us to just fail. We don't have to (and aren't) using this verifier for any of our stuff.